src/test/prettyPrint/expressions.spec.ts

```diff
@@ -0,0 +1,119 @@
+import fc from "fast-check";
+import {
+    AstConditional,
+    AstExpression,
+    AstNumber,
+    AstOpBinary,
+    AstOpUnary,
+    eqExpressions,
+    getAstFactory,
+} from "../../grammar/ast";
+import { prettyPrint } from "../../prettyPrinter";
+import { dummySrcInfo, getParser } from "../../grammar";
+
+describe("Pretty Print Expressions", () => {
+    // Max depth of the expression tree
+    const maxShrinks = 15;
+
+    const generateAstNumber = () =>
+        fc.record({
+            kind: fc.constant("number"),
+            base: fc.constantFrom(2, 8, 10, 16),
+            value: fc.bigInt().filter((n) => n > 0),
+            id: fc.constant(0),
+            loc: fc.constant(dummySrcInfo),
+        }) as fc.Arbitrary<AstNumber>;
+
+    const generateAstOpUnary = (expression: fc.Arbitrary<AstExpression>) =>
+        fc.record({
+            kind: fc.constant("op_unary"),
+            op: fc.constantFrom("+", "-", "!", "!!", "~"),
+            operand: expression,
+            id: fc.constant(0),
+            loc: fc.constant(dummySrcInfo),
+        }) as fc.Arbitrary<AstOpUnary>;
+
+    const generateAstOpBinary = (expression: fc.Arbitrary<AstExpression>) =>
+        fc.record({
+            kind: fc.constant("op_binary"),
+            op: fc.constantFrom(
+                "+",
+                "-",
+                "*",
+                "/",
+                "!=",
+                ">",
+                "<",
+                ">=",
+                "<=",
+                "==",
+                "&&",
+                "||",
+                "%",
+                "<<",
+                ">>",
+                "&",
+                "|",
+                "^",
+            ),
+            left: expression,
+            right: expression,
+            id: fc.constant(0),
+            loc: fc.constant(dummySrcInfo),
+        }) as fc.Arbitrary<AstOpBinary>;
+
+    const generateAstConditional = (expression: fc.Arbitrary<AstExpression>) =>
+        fc.record({
+            kind: fc.constant("conditional"),
+            condition: expression,
+            thenBranch: expression,
+            elseBranch: expression,
```

Comment:
These need to be 3 different expressions

Reply:
Fixed, every expression is new-generated